extractor/src/main/java/org/schabi/newpipe/extractor/channel/ChannelTabExtractor.java

```diff
@@ -0,0 +1,12 @@
+package org.schabi.newpipe.extractor.channel;
+
+import org.schabi.newpipe.extractor.InfoItem;
+import org.schabi.newpipe.extractor.ListExtractor;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+
+public abstract class ChannelTabExtractor extends ListExtractor<InfoItem> {
```

Comment:
Shouldn't `getName()` represent the name of an item? In this case if I understand correctly it is being used only to pass information about the tab to the app. For that purpose using a shared enum or some integer constants would be a better fit, since those can't contain typos. So I would add a `getTabType()`  base function here. In case that's not done, I'd suggest adding a notice here about the different usage of `getName()`.

Reply:
Isn't e.g. "Videos" the name of the tab?